scripts/build-windows

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+set -e -x
+
+cd $(dirname $0)/..
+
+. ./scripts/version.sh
+
+GO=${GO-go}
+
+PKG="github.com/k3s-io/k3s"
+PKG_CONTAINERD="github.com/containerd/containerd"
+PKG_K3S_CONTAINERD="github.com/k3s-io/containerd"
+PKG_CRICTL="github.com/kubernetes-sigs/cri-tools/pkg"
+PKG_K8S_BASE="k8s.io/component-base"
+PKG_K8S_CLIENT="k8s.io/client-go/pkg"
+
+buildDate=$(date -u '+%Y-%m-%dT%H:%M:%SZ')
+
+VERSIONFLAGS="
+    -X ${PKG}/pkg/version.Version=${VERSION}
+    -X ${PKG}/pkg/version.GitCommit=${COMMIT:0:8}
+
+    -X ${PKG_K8S_CLIENT}/version.gitVersion=${VERSION}
+    -X ${PKG_K8S_CLIENT}/version.gitCommit=${COMMIT}
+    -X ${PKG_K8S_CLIENT}/version.gitTreeState=${TREE_STATE}
+    -X ${PKG_K8S_CLIENT}/version.buildDate=${buildDate}
+
+    -X ${PKG_K8S_BASE}/version.gitVersion=${VERSION}
+    -X ${PKG_K8S_BASE}/version.gitCommit=${COMMIT}
+    -X ${PKG_K8S_BASE}/version.gitTreeState=${TREE_STATE}
+    -X ${PKG_K8S_BASE}/version.buildDate=${buildDate}
+
+    -X ${PKG_CRICTL}/version.Version=${VERSION_CRICTL}
+
+    -X ${PKG_CONTAINERD}/version.Version=${VERSION_CONTAINERD}
+    -X ${PKG_CONTAINERD}/version.Package=${PKG_K3S_CONTAINERD}
+
+    -X ${PKG_CONTAINERD}/version.Version=${VERSION_CONTAINERD}
+    -X ${PKG_CONTAINERD}/version.Package=${PKG_K3S_CONTAINERD}
+"
+
+LDFLAGS="
+    -w -s"
+
+STATIC="-extldflags '-static'"
+# PR: Do we need the following linker flags for Windows binary?
+# STATIC="
+#     -extldflags '-static -lm -ldl -lz -lpthread'
```

Comment:
IIRC these are all required to build a static binary, but since we're not shipping the resulting binary, just making sure that it builds, dropping these is probably fine.

Reply:
I agree wit you. BTW I downloaded and grep'ed RKE2 windows build and there isn't any of these linker flags there.

```sh
grep -B5 -A10 "bin/rke2.exe -ldflags" [a temp dir]/logs_rancher_rke2_3007_1_2.log
+ echo
+ GOOS=windows
+ CGO_ENABLED=1
+ CXX=x86_64-w64-mingw32-g++
+ CC=x86_64-w64-mingw32-gcc
+ go build -tags ' netgo osusergo no_stage static_build sqlite_omit_load_extension ' -o bin/rke2.exe -ldflags '  -w 
        -X github.com/k3s-io/k3s/pkg/version.GitCommit=342347537760e72cef347ddb2dcbdcbb35754e30
        -X github.com/k3s-io/k3s/pkg/version.Program=rke2
        -X github.com/k3s-io/k3s/pkg/version.Version=v1.23.4-dev+e1aeae66
        -X github.com/rancher/rke2/pkg/images.DefaultRegistry=docker.io
        -X github.com/rancher/rke2/pkg/images.DefaultEtcdImage=rancher/hardened-etcd:v3.5.1-k3s1-build20220112
        -X github.com/rancher/rke2/pkg/images.DefaultKubernetesImage=rancher/hardened-kubernetes:v1.23.4-rke2r1-build20220217
        -X github.com/rancher/rke2/pkg/images.DefaultPauseImage=rancher/pause:3.6
        -X github.com/rancher/rke2/pkg/images.DefaultRuntimeImage=rancher/rke2-runtime:v1.23.4-dev-e1aeae66
'
```

Reply:
RKE2 on Windows doesn't have the same static/hardened requirements as the Linux builds. So yeah, I think it's fine.